### ISSUE TITLE
rollup: wrong output directory if multiple inputs (windows only)

### ISSUE THREAD

**Issue by manuel3108** [unplugin/unplugin-isolated-decl#10]
### Describe the bug

On windows (! important !) rollup builds that have multiple inputs will be generated in the wrong directory. This can best be seen here: 
![image](https://github.com/user-attachments/assets/f99c8101-b288-4b0a-b53f-5f06b028661c)

They are nested two layers deeper than they should. In comparison i created a project with only one input, where everything works as expected.

**Only reproducible on windows**. Seems to be working on linux as expected, even from the `wsl`

To reproduce:
* checkout the repo linked below
* `pnpm i`
* `pnpm build`
* and have a look at both generated `dist` directories

### Reproduction

https://github.com/manuel3108/dts-reproduction.git

### System Info

```Shell
System:
    OS: Windows 11 10.0.22631
    CPU: (12) x64 AMD Ryzen 5 2600 Six-Core Processor
    Memory: 6.89 GB / 15.93 GB
  Binaries:
    Node: 22.6.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.8.2 - C:\Program Files\nodejs\npm.CMD
    pnpm: 9.8.0 - ~\AppData\Roaming\npm\pnpm.CMD
    bun: 1.1.22 - ~\.bun\bin\bun.EXE
  Browsers:
    Edge: Chromium (127.0.2651.98)
    Internet Explorer: 11.0.22621.3527
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by manuel3108** [comment]
@AdrianGonz97 found out that replacing this

```js
    const inputs = [`./packages/${project}/index.ts`];
    const outDir = `./packages/${project}/dist`;

    if (project === 'two-inputs') inputs.push(`./packages/${project}/cli.ts`);
```

inside `rollup.config.js` with 

```js
    const inputs = [path.resolve("packages", project, "index.ts")];
    const outDir = path.resolve("packages", project, "dist");

    if (project === "two-inputs")
        inputs.push(path.resolve("packages", project, "cli.ts"));
```

seems to solve the problem.

Though it's still unclear why this is only happening on windows, and only relevant if the package has two inputs

**Comment by sxzz** [comment]
Try `pnpm i https://pkg.pr.new/unplugin/unplugin-isolated-decl@353d87b`. If it resolves the issue, I'll release a new version.

**Comment by manuel3108** [comment]
Fantastic work, that fixes the issue!

**Comment by sxzz** [comment]
Released as `unplugin-isolated-decl@0.4.7`
